spec/inputs/tcp_spec.rb

```diff
@@ -77,11 +77,11 @@
 
     events = input(conf) do |pipeline, queue|
       socket = Stud::try(5.times) { TCPSocket.new("127.0.0.1", port) }
-      socket.puts("PROXY TCP4 1.2.3.4 5.6.7.8 1234 5678\r");
+      socket.syswrite("PROXY TCP4 1.2.3.4 5.6.7.8 1234 5678\r\n")
       socket.flush
       event_count.times do |i|
         # unicode smiley for testing unicode support!
-        socket.puts("#{i} ☹")
+        socket.syswrite("#{i} ☹\n")
```

Comment:
If you are switching to `syswrite` (instead of `puts`) you do not need to call `flush` (next line down)